enzy_htp/analysis/rmsd.py

```diff
@@ -0,0 +1,120 @@
+    return mask_pattern
+
+def rmsd_of_structure(structure_ensemble: StructureEnsemble, interface: str = "amber", reference_structure: Structure = None, include_ligand: bool = True, ca_only: bool = True) -> float:
+    """Get RMSD value from MD simulation result of the whole structure.
+    
+    Args:
+        structure_ensemble (StructureEnsemble): A collection of different geometries of the same enzyme structure.
+        interface (str, optional): The interface for calculation. Default amber.
+        reference_structure (Structure, optional): The reference structure, which should have the same sequences and ligangs as the structure_ensemble. Default: Average structure.
+        include_ligand (bool, optional): Indicate if ligands are included in RMSD calculation. Default True.
+        ca_only (bool, optional): Indicate if only C-alpha are included in RMSD calculation; otherwise all atoms except hydrogens are included. Default True.
+    """
+    interface = interface.lower()
+    mask_region = structure_ensemble.structure_0.residues
+    if (not include_ligand):
+        mask_region = list(filter(lambda residue: (not residue.is_noncanonical()), mask_region))
+    return rmsd_of_region(
+        structure_ensemble=structure_ensemble, 
+        interface=interface,
+        reference_structure=reference_structure, 
+        mask_region=mask_region, 
+        ca_only=ca_only
+    )
+
+def rmsd_of_region(structure_ensemble: StructureEnsemble, interface: str = "amber", reference_structure: Structure = None, mask_region: List[Residue] = list(), ca_only: bool = True) -> float:
+    """Get RMSD value from MD simulation result with a mask region (a list of residues).
+    
+    Args:
+        structure_ensemble (StructureEnsemble): A collection of different geometries of the same enzyme structure.
+        interface (str, optional): The interface for calculation. Default amber.
+        reference_structure (Structure, optional): The reference structure, which should have the same sequences and ligangs as the structure_ensemble. Default: Average structure.
+        mask_region (List[Residue], optional): A list of Residues to calculation RMSD value. If the list is empty, then all residues are selected.
+        ca_only (bool, optional): Indicate if only C-alpha are included in RMSD calculation; otherwise all atoms except hydrogens are included. Default True.
+    """
+    interface = interface.lower()
+    mask_pattern = compose_mask_pattern(mask_region=mask_region, interface=interface, ca_only=ca_only)
+    return rmsd_with_pattern(
+        structure_ensemble=structure_ensemble, 
+        interface=interface,
+        reference_structure=reference_structure, 
+        mask_pattern=mask_pattern
+    )
+
+def rmsd_with_pattern(structure_ensemble: StructureEnsemble, interface: str = "amber", reference_structure: Structure = None, mask_pattern: str = str()) -> float:
```

Comment:
This is the most reasonable API. However the argument names are barely readable. Make it
```
def rmsd(
    stru_esm,
    region_pattern,
    ref_stru,
    engine,
)
```
Always put the I/O (e.g.: "stru_esm") before the configurative (e.g.: "engine")

Reply:
"mask" and "pattern" means the same thing fyi